From c99e1d0c91ac8d7db3062ea1af315f21295701d7 Mon Sep 17 00:00:00 2001
From: Chuhong Yuan <hslester96@gmail.com>
Date: Tue, 9 Jun 2020 00:22:26 +0800
Subject: [PATCH] mmc: sdhci-of-arasan: Add missed checks for
 devm_clk_register()
Git-commit: c99e1d0c91ac8d7db3062ea1af315f21295701d7
Patch-mainline: v5.9-rc1
References: git-fixes

These functions do not check the return value of devm_clk_register():
  - sdhci_arasan_register_sdcardclk()
  - sdhci_arasan_register_sampleclk()

Therefore, add the missed checks to fix them.

Fixes: c390f2110adf1 ("mmc: sdhci-of-arasan: Add ability to export card clock")
Signed-off-by: Chuhong Yuan <hslester96@gmail.com>
Reviewed-by: Douglas Anderson <dianders@chromium.org>
Link: https://lore.kernel.org/r/20200608162226.3259186-1-hslester96@gmail.com
Signed-off-by: Ulf Hansson <ulf.hansson@linaro.org>
Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/mmc/host/sdhci-of-arasan.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/mmc/host/sdhci-of-arasan.c
+++ b/drivers/mmc/host/sdhci-of-arasan.c
@@ -528,6 +528,8 @@ static int sdhci_arasan_register_sdclk(s
 	sdhci_arasan->sdcardclk_hw.init = &sdcardclk_init;
 	sdhci_arasan->sdcardclk =
 		devm_clk_register(dev, &sdhci_arasan->sdcardclk_hw);
+	if (IS_ERR(sdhci_arasan->sdcardclk))
+		return PTR_ERR(sdhci_arasan->sdcardclk);
 	sdhci_arasan->sdcardclk_hw.init = NULL;
 
 	ret = of_clk_add_provider(np, of_clk_src_simple_get,
